metashare/api/models.py

```diff
@@ -204,20 +223,89 @@ class TaskSlug(AbstractSlug):
 
 
 class Task(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Model):
+    tracker = FieldTracker(fields=("branch_name",))
+
     name = models.CharField(max_length=50)
     project = models.ForeignKey(Project, on_delete=models.PROTECT, related_name="tasks")
     description = MarkdownField(blank=True, property_suffix="_markdown")
     assignee = models.ForeignKey(
         User, on_delete=models.SET_NULL, null=True, related_name="assigned_tasks"
     )
+    branch_name = models.SlugField(max_length=50, null=True)
 
     slug_class = TaskSlug
 
+    def __str__(self):
+        return self.name
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+
+        if self.tracker.has_changed("branch_name"):
+            self.notify_has_branch_name()
+
+    def notify_has_branch_name(self):
+        from .serializers import TaskSerializer
+
+        payload = TaskSerializer(self).data
+        message = {"type": "TASK_UPDATE", "payload": payload}
+        async_to_sync(push.push_message_about_instance)(self, message)
+
     class Meta:
         ordering = ("-created_at", "name")
         unique_together = (("name", "project"),)
 
 
+SCRATCH_ORG_TYPES = Choices("Dev", "QA")
+
+
+class ScratchOrg(mixins.HashIdMixin, mixins.TimestampsMixin, models.Model):
+    tracker = FieldTracker(fields=("url",))
+
+    task = models.ForeignKey(Task, on_delete=models.PROTECT)
+    org_type = StringField(choices=SCRATCH_ORG_TYPES)
+    owner = models.ForeignKey(User, on_delete=models.PROTECT)
+    last_modified_at = models.DateTimeField(null=True)
+    expires_at = models.DateTimeField(null=True)
+    latest_commit = StringField(blank=True)
+    latest_commit_url = models.URLField(blank=True)
+    url = models.URLField(null=True)
+    has_changes = models.BooleanField(default=False)
+
+    def subscribable_by(self, user):  # pragma: nocover
+        # TODO: revisit this?
+        return True
+
+    def save(self, *args, **kwargs):
+        create_remote_resources = self.id is None
+        super().save(*args, **kwargs)
+
+        if create_remote_resources:
+            self.create_remote_resources()
```

Comment:
It no longer seems like this is happening out-of-band. That is, the `POST` is just pending until the entire job finishes, then returns with a `201`.

Reply:
Fixed in https://github.com/SFDO-Tooling/MetaShare/pull/142/commits/ab2670b6553a54cb0420c12df09adbedd47593ab